LevelSpaceXWGUI.nlogo

```diff
@@ -927,7 +933,15 @@ to-report get-from-model-all-types [model-id a-type]
 end
 
 to-report  get-eligible-arguments [an-entity]
-  report filter [table:get last ? "type" = "reporter" and table:get last ? "args" = [] ] table:to-list tasks
+  let observer? table:get an-entity "type" = "observer"
+  let model table:get an-entity "model"
+  report filter [
+    table:get last ? "type" = "reporter" and
+    table:get last ? "args" = [] and
+    (not observer? or member? "O" table:get last ? "contexts") and
+    (model = table:get last ? "model" or member? "O" table:get last ? "contexts") and
```

Comment:
I'm not really clear on why arguments are selected, but this is a very long predicate to filter a list on. Do you think it would be clearer to break it out into a separate reporter?